packages/flutter/lib/src/material/material.dart

```diff
@@ -86,9 +104,18 @@ abstract class MaterialInkController {
 /// [InkSplash] and [InkHighlight] effects. To trigger a reaction on the
 /// material, use a [MaterialInkController] obtained via [Material.of].
 ///
-/// If a material has a non-zero [elevation], then the material will clip its
-/// contents because content that is conceptually printing on a separate piece
-/// of material cannot be printed beyond the bounds of the material.
+/// ## Shape
+/// The shape for material is determined by [type] and [borderRadius].
+///  - If [borderRadius] is non null, the shape is a rounded rectangle, with
+///    corners specified by [borderRadius].
+///  - If [borderRadius] is null, [type] determines the shape as follows:
+///    - [MaterialType.canvas]: the default material shape is a rectangle.
+///    - [MaterialType.card]: the default material shape is a rectangle with
+///      rounded edges. The edge radii is specified by [kMaterialEdges].
+///    - [MaterialType.circle]: the default material shape is a circle.
+///    - [MaterialType.button]: the default material shape is a rectangle with
+///      rounded edges. The edge radii is specified by [kMaterialEdges].
+///    - [MaterialType.transparency]: the default material shape is a rectangle.
 ///
 /// If the layout changes (e.g. because there's a list on the material, and it's
```

Comment:
this needs a section header too (it's not about shapes)

Reply:
done